lib/Parse/ParseExpr.cpp

```diff
@@ -2949,6 +3063,9 @@ ExprResult Parser::DeferredParseBoundsExpression(std::unique_ptr<CachedTokens> T
   ConsumeAnyToken();   // Skip past the current token to the new tokens.
   ExprResult Result = ParseBoundsExpression();
 
+  if (StartsRelativeBoundsClause(Tok))
```

Comment:
I think this code needs to be changed  a bit.    We have a stream of tokens have the same "shape" as a bounds expression followed by relative alignment clause, but that may not actually be valid bounds expressions or relative alignment clauses.   There are four possible cases here:

1.  Result is invalid.
2.  The tok is not EOF and starts a relative bounds clause
3.  The tok is not EOF and does not start a relative bounds clause.
4. The tok is EOF.

For case 1, we want to consume the remaining tokens.
For case 2, we want to parse the relative alignment clause.  If there are any tokens left after parsing the relative alignment clause, we want to consume them.    We should have generated an error message while parsing the relative alignment clause.
For case 3, we need to issue an error message and then consume the rest of the tokens.
For case 4, we do not need to do anything.